x-pack/plugins/maps/public/routing/state_syncing/app_state_manager.ts

```diff
@@ -5,15 +5,24 @@
  */
 
 import { Subject } from 'rxjs';
+import { Filter, SavedQuery, Query } from 'src/plugins/data/public';
 
 export class AppStateManager {
-  _query = '';
-  _savedQuery = '';
-  _filters = [];
+  _query: string | { [key: string]: any } = '';
+  _savedQuery: SavedQuery | string = '';
```

Comment:
Instead of defaulting to `''`, initial value should be 
```
{
    query: '',
    language: getUiSettings().get('search:queryLanguage'),
}
```

This way the type can just be `SavedQuery`

Reply:
@nreese The `SavedQuery` interface looks like this. I couldn't find `query` and `language` in that type. Am I looking in the wrong place?
```
export interface SavedQuery {
  id: string;
  attributes: SavedQueryAttributes;
}
```

Reply:
@shamin, you are corrrect. Sorry for the confusion. The provided type does not match `SavedQuery`.

Digging a bit deeper, I have found the source of the problem. During the conversion to New Platform, [fetching saved query from the savedQueryId](https://github.com/elastic/kibana/blob/7.8/x-pack/legacy/plugins/maps/public/angular/map_controller.js#L278) was lost. This is the source of the typing errors.

This bug does not need to be resolved in this PR. We can fix later, after this PR has merged.

Please do the following:
* In map_app_view.ts
  * state interface, savedQuery should be `SavedQuery`
  * in `componentDidMount`, comment out the call to `_updateStateFromSavedQuery` and add a comment that savedQuery must be fetched from savedQueryId
  * type _updateStateFromSavedQuery as `_updateStateFromSavedQuery = (savedQuery: SavedQuery)` 
* In app_state_manager.ts
  * Can you rename `_savedQuery` to `_savedQueryId` and type as string
  * Can you rename `savedQuery` parameter in `setQueryAndFilters` to `savedQueryId`


This should clear up all of the typing and then we can fix the bug external to this PR. Great to see typescript highlight some of these underlying problems.

Reply:
@nreese One small issue. As we changed the parameters in `setQueryAndFilters` function, We need to update this [function call](https://github.com/elastic/kibana/pull/76387/files/765f772c4ea358eb9fff43f2ea2558239d507a17#diff-2883fe5bc7ce631c1d2598f5c4c8ba9fR327) to send savedQueryId, so it should look like this right? 
```
this._appStateManager.setQueryAndFilters({ savedQueryId: savedQuery.id });
```

Reply:
that makes sense.

Reply:
Thanks, Made the change.